### PR TITLE
NSNotificationCenter observers synchronization

### PR DESCRIPTION
**Pull request by ochococo**

### CODE REVIEW

File: Foundation/NSNotification.swift [hunk 1]
```diff
@@ -105,20 +140,17 @@ public class NSNotificationCenter : NSObject {
...
-        }
-    }
 
+        _observersLock.synchronized({
+            self._observers = _observers.filterOutObserver(observer, name: name, object: object)
+        })
+    }
     
-    public func addObserverForName(name: String?, object obj: AnyObject?, queue: NSOperationQueue?, usingBlock block: (NSNotification) -> Void) -> NSObjectProtocol {
+    public func addObserverForName(name: String?, object obj: AnyObject?, queue: NSOperationQueue?, usingBlock block: (NSNotification) -> ()) -> NSObjectProtocol {
```
Comment (phausler) [inline Foundation/NSNotification.swift]: why the change to ()? Void seems more readable/appropriate to me

Comment (ochococo) [inline Foundation/NSNotification.swift]: You're right. It's more readable now. Fixed ✅
